src/Symfony/Component/Security/Core/CHANGELOG.md

```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+7.3
+---
+
+ * Deprecate `UserInterface::eraseCredentials()`, use a dedicated DTO or erase credentials
+   on your own e.g. upon `AuthenticationTokenCreatedEvent` instead
```

Comment:
Would it be possible to use `__serialize()`/`__deserialize()` instead to only serialize the intended fields? Or is this something we don't want to encourage?

Reply:
Good point, I think the DTO approach is the best but the serialization one is also valid https://stackoverflow.com/questions/42074225/symfony-userinterface-is-serializing-the-entire-massive-user-entity.  I'm going to update the deprecation messages and upgrade/changelog instructions to mention it, thanks.

Reply:
Thinking more about this, I think we should not suggest it: indefinitely keeping plain credentials on this object is likely to lead to bad situations e.g. having credentials persisted in database or exposed elsewhere e.g. logs